### ISSUE TITLE
Chapter 7: context processors does not work

### ISSUE THREAD

**Issue by Rahimz** [PacktPublishing/Django-3-by-Example#62]
When I add this line in my settings to use context processors for using cart info in all templates:
```
'context_processors': [
#....
'cart.context_processors.cart',
```
I get **ModuleNotFoundError**. It says: **No module named 'cart.context_processors'**.

Does anyone see this error?

**Comment by qodatecnologia** [comment]
Yeah! i solved that! the book has a too much bad code with syntax errors. On of them is the name of context processors. In book: "context_ processors.py" but the right is "context_processors.py" without that space after underscore. Try this!! and be careful with the codes: too many basic errors

**Comment by lukstaron** [comment]
Hi Antonio, I make the shop on django 3.2 version (as 3.0.9 is no longer supported), I am not getting this context processor to work - despite i did everything by the book, I am getting 'Your cart is empty' in the template despite adding items to the cart.

I solved previous error with the wrong filename but here I am not getting why....

This code is so simple :) what can go wrong?

Settings: 
`TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'cart.context_processors.cart',
            ],
        },
    },
]
`Shop/templates/shop/base.html:

`<div class="cart"> 
            {% with total_items=cart|length %}
            {% if total_items > 0 %}
              Your cart:
              <a href="{% url "cart:cart_detail" %}">
                {{ total_items }} item{{ total_items|pluralize }},
                ${{ cart.get_total_price }}
              </a>
            {% else %}
              Your cart is empty.
            {% endif %}
          {% endwith %}
        </div>`
cart/context_processors.py:
`from .cart import Cart


def cart(request):
    return {'cart': Cart(request)}`
